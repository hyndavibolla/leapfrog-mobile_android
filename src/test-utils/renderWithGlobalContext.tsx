import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
