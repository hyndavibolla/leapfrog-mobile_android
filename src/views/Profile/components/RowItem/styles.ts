import { css } from '@emotion/native';
import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants/styles';

export const styles = {
  linksCard: css`
    background-color: ${COLOR.WHITE};
    margin-bottom: 8px;
    flex-direction: row;
  `,
  linksText: css`
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.SMALL};
    color: ${COLOR.BLACK};
  `,
  iconContainer: css`
    align-items: center;
    justify-content: center;
  `,
  rightContent: css`
    flex-direction: row;
    align-items: center;
  `,
  valueText: css`
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.PETITE};
    line-height: ${FONT_SIZE.BIG};
    font-weight: 500;
    color: ${COLOR.DARK_GRAY};
    margin-right: 8px;
  `
};
