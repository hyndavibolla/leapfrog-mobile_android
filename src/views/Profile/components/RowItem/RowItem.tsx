import React, { memo, ReactElement } from 'react';
import { TouchableHighlight, Text, View } from 'react-native';

import { useTestingHelper } from '_utils/useTestingHelper';
import { Card } from '_components/Card/Card';

import { styles } from './styles';

export interface Props {
  name: string;
  onPress: () => void;
  testIdName: string;
  icon: ReactElement;
  value?: string;
}

export const RowItem = memo(({ name, onPress, testIdName, icon, value }: Props) => {
  const { getTestIdProps } = useTestingHelper('profile-row-item');
  return (
    <TouchableHighlight onPress={onPress} underlayColor="transparent" activeOpacity={1} {...getTestIdProps(testIdName)}>
      <Card style={styles.linksCard}>
        <Text style={styles.linksText}>{name}</Text>
        <View style={styles.rightContent}>
          {value ? (
            <Text style={styles.valueText} {...getTestIdProps(`${testIdName}-value`)}>
              {value}
            </Text>
          ) : null}
          <View style={styles.iconContainer}>{icon}</View>
        </View>
      </Card>
    </TouchableHighlight>
  );
});
