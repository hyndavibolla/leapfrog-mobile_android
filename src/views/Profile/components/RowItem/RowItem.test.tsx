import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { RowItem, Props } from './RowItem';
import { wait } from '_utils/wait';
import GoOutIcon from '_assets/shared/goOutIcon.svg';

describe('RowItem', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      name: 'Edit your profile',
      onPress: jest.fn(),
      testIdName: 'edit-your-profile',
      icon: <GoOutIcon />
    };
  });

  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<RowItem {...props} />);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be pressable', async () => {
    const { getByTestId } = renderWithGlobalContext(<RowItem {...props} />);
    fireEvent.press(getByTestId('profile-row-item-edit-your-profile'));
    expect(props.onPress).toBeCalled();
  });
});
