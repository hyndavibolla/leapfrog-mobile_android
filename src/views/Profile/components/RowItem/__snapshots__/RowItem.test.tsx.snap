// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowItem should render 1`] = `
<View
  accessibilityLabel="profile-row-item-edit-your-profile"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="profile-row-item-edit-your-profile"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 18,
        }
      }
    >
      Edit your profile
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        GoOutIcon
      </View>
    </View>
  </View>
</View>
`;
